lib/Driver/CompilationRecord.h

```diff
@@ -0,0 +1,80 @@
+//===--- CompilationRecord.h ------------------------------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_DRIVER_COMPILATIONRECORD_H
+#define SWIFT_DRIVER_COMPILATIONRECORD_H
+
+#include "swift/Driver/Action.h"
+
+namespace swift {
+namespace driver {
+namespace compilation_record {
+
+/// Compilation record files (.swiftdeps files) are YAML files composed of these
+/// top-level keys.
+enum class TopLevelKey {
+  /// The key for the Swift compiler version used to produce the compilation
+  /// record.
+  Version,
+  /// The key for the list of arguments passed to the Swift compiler when
+  /// producing the compilation record.
+  Options,
+  /// The key for the time at which the build that produced the compilation
+  /// record started.
+  BuildTime,
+  /// The key for the list of inputs to the compilation that produced the
+  /// compilation record.
+  Inputs,
+};
+
+/// \returns A string representation of the given key.
+inline static StringRef getName(TopLevelKey Key) {
+  switch (Key) {
+  case TopLevelKey::Version: return "version";
+  case TopLevelKey::Options: return "options";
+  case TopLevelKey::BuildTime: return "build_time";
+  case TopLevelKey::Inputs: return "inputs";
+  }
+}
+
+/// \returns The string identifier used to represent the given status in a
+/// compilation record file (.swiftdeps file).
```

Comment:
Ah, I was actually confused when you mentioned that earlier. By "not round-trip", you mean that `getInfoStatusForIdentifier(getIdentifierForInputInfoStatus(Status)) != Status`, is that right? Since the left-hand side would be an `Optional<Status>`?